source/slang/slang-ir-diff-jvp.cpp

```diff
@@ -212,14 +244,14 @@ struct JVPTranscriber
     {
         IRInst* storeLocation = storeP->getPtr();
         IRInst* storeVal = storeP->getVal();
-        if (auto destParam = as<IRVar>(storeLocation))
+        if (as<IRVar>(storeLocation) || as<IRParam>(storeLocation))
         {   
             // If the stored value has a differential version, 
             // emit a store instruction for the differential parameter.
             // Otherwise, emit nothing since there's nothing to load.
             // 
             IRInst* storeValD = getDifferentialInst(storeVal);
-            IRVar*  storeLocationD = as<IRVar>(getDifferentialInst(destParam));
+            IRInst* storeLocationD = getDifferentialInst(storeLocation);
```

Comment:
Does `getDifferentialInst` always returns a pointer-type value? I mention this because there is a difference between an `IRVar` and an `IRParam`. An `IRVar` is always a pointer, but an `IRParam` is just an SSA temporary value, so you should never need to load an `IRParam`.

Reply:
Yeah, you shouldn't load a `IRParam`.
For example, if I have func(int p);
Then `p` is just `p = IRParam : int`.
If I have `int i;` as an variable, then the IR is
`i = IRVar:Ptr<int>`, and you need to load it to get an `int`.